0.2.4 & master fail to compile against newest glib / gnome 3.24

It looks like one of the used GLib functions changed it's API, and compilation of audience fails with this error:

```
(...)/src/Audience.vala:62.13-62.26: error: 1 missing arguments for `void GLib.Application.set_default (GLib.Application?)'
            set_default ();
            ^^^^^^^^^^^^^^
make[2]: *** [src/CMakeFiles/audience.dir/build.make:164: src/audience_valac.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:152: src/CMakeFiles/audience.dir/all] Error 2
```

--- a/src/Audience.vala
+++ b/src/Audience.vala
@@ -59,7 +59,6 @@ namespace Audience {
             Gtk.Settings.get_default ().gtk_application_prefer_dark_theme = true;
             this.flags |= GLib.ApplicationFlags.HANDLES_OPEN;
             settings = new Settings ();
-            set_default ();
         }
 
         private static App app; // global App instance

